evalml/tests/utils_tests/test_nullable_type_utils.py

```diff
@@ -0,0 +1,223 @@
+        data,
+        handle_boolean_nullable=True,
+        handle_integer_nullable=True,
+    )
+
+    assert data_d is data
+
+
+@pytest.mark.parametrize("has_nans", [True, False])
+@pytest.mark.parametrize("bool_null_incompatible", [True, False])
+@pytest.mark.parametrize("int_null_incompatible", [True, False])
+def test_downcast_nullable_X_replaces_nullable_types(
+    split_nullable_logical_types_by_compatibility,
+    nullable_type_test_data,
+    int_null_incompatible,
+    bool_null_incompatible,
+    has_nans,
+):
+    X = nullable_type_test_data(has_nans=has_nans)
+    # Set other typing info to confirm it's maintained
+    X.ww.init(
+        schema=X.ww.schema,
+        column_origins={"int col nullable": "base", "float col": "engineered"},
+    )
+    original_X = X.ww.copy()
+
+    # Confirm nullable types are all present in original data
+    assert (
+        len(
+            original_X.ww.select(
+                ["IntegerNullable", "BooleanNullable", "AgeNullable"],
+            ).columns,
+        )
+        > 0
+    )
+
+    _, incompatible_ltypes = split_nullable_logical_types_by_compatibility(
+        int_null_incompatible,
+        bool_null_incompatible,
+    )
+
+    compatible_original_schema = original_X.ww.select(
+        exclude=incompatible_ltypes,
+        return_schema=True,
+    )
+
+    X_d = _downcast_nullable_X(
+        X,
+        handle_boolean_nullable=bool_null_incompatible,
+        handle_integer_nullable=int_null_incompatible,
+    )
+
+    assert set(X_d.columns) == set(original_X.columns)
+    assert len(X_d.ww.select(incompatible_ltypes).columns) == 0
+    assert X_d.ww["int col nullable"].ww.origin == "base"
+
+    # Confirm the other columns' woodwork info is unchanged
+    undowncasted_partial_schema = original_X.ww.get_subset_schema(
+        compatible_original_schema.columns.keys(),
+    )
+    assert compatible_original_schema == undowncasted_partial_schema
+
+
+@pytest.mark.parametrize("has_nans", [True, False])
+@pytest.mark.parametrize(
+    "nullable_ltype",
+    ["BooleanNullable", "IntegerNullable", "AgeNullable"],
+)
+@pytest.mark.parametrize("bool_null_incompatible", [True, False])
+@pytest.mark.parametrize("int_null_incompatible", [True, False])
+def test_downcast_nullable_y_replaces_nullable_types(
+    split_nullable_logical_types_by_compatibility,
+    nullable_type_target,
+    int_null_incompatible,
+    bool_null_incompatible,
+    nullable_ltype,
+    has_nans,
+):
+    y = nullable_type_target(ltype=nullable_ltype, has_nans=has_nans)
+
+    y_d = _downcast_nullable_y(
+        y,
+        handle_boolean_nullable=bool_null_incompatible,
+        handle_integer_nullable=int_null_incompatible,
+    )
+
+    (
+        y_compatible_ltypes,
+        y_incompatible_ltypes,
+    ) = split_nullable_logical_types_by_compatibility(
+        int_null_incompatible,
+        bool_null_incompatible,
+    )
+
+    if isinstance(nullable_ltype, tuple(y_compatible_ltypes)):
```

Comment:
can't seem to figure it out but lets make sure we're hitting this case before merging!

Reply:
So turns out `nullable_ltype` is a string so it'll never be an instance of the compatible ltypes 🤦‍♀️ 

Fixed that there and in the component and objective tests as well as a couple of other small bugs in these checks. 878cfc24c